src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/Analizador.kt

```diff
@@ -0,0 +1,73 @@
+package ar.edu.unahur.obj2.servidorWeb
+
+import java.time.LocalDateTime
+
+/* Se crea una clase Analizador que se va a usar de base para los 3 modelos necesarios. */
+abstract class Analizador() {
+    abstract fun analizarRespuesta(respuesta: Respuesta)
+}
+
+/* Se crea un analizador que evalua la demora de los pedidos     */
+/* Si el tiempo en la respuesta es mayor que el tiempo de demora */
+/* el pedido se agrega en una variable Demoradas                 */
+class AnalizadorDeDemora(val tiempoDeDemora: Int):Analizador() {
+    val Demoradas = mutableListOf<Respuesta>()
+
+    override fun analizarRespuesta(respuesta: Respuesta) {
+        if (respuesta.tiempo > tiempoDeDemora) {
+            this.Demoradas.add(respuesta)
+        }
+    }
+    fun cantidadRespuestasDemoradas() = this.Demoradas.size
+}
+
+/* Se crea un analizador que evalua si los pedidos fueron realizados por una IP sospechosa      */
+/* Se evalua una respuesta, y si fue realizada por una IP sospechosa, se agrega en una variable */
+/* IPsSospechosas. Ademas se guarda el modulo que creo la respuesta en una variable Modulos     */
+class AnalizadorDeIPSospechosas(val IPsSospechosas : List<String>):Analizador() {
+    val pedidosSospechosos = mutableListOf<Pedido>()
+    val modulos = mutableListOf<Modulo?>()
+
+    override fun analizarRespuesta(respuesta: Respuesta){
+        if (IPsSospechosas.contains(respuesta.pedido.ip)){
+            this.pedidosSospechosos.add(respuesta.pedido)
+            }
+        modulos.add(respuesta.moduloCreador)
+    }
+    /* Devuelve cuantos pedidos realizo una IP sospechosa */
+    fun pedidosRealizados(ip: String) = this.pedidosSospechosos.count{ it.ip == ip }
+
+    /* Devuelve cual fue el modulo mas consultado por determinada IP */
+    fun moduloMasConsultado() = modulos.groupBy {it}.mapValues {it.value.size}.maxBy {it.value}?.key
+
+    /* Devuelve que IPs utilizaron determinada ruta */
+    fun rutaRequeridaPorIPS(ruta: String): List<String>  = pedidosSospechosos.filter {it.rutaDePedido() == ruta}.map {it.ip}
+
+}
+
+/* Se crea un analizador de estadisticas. Que analiza las respuestas para crear estadisticas */
+class AnalizadorDeEstadisticas():Analizador() {
+    val respuestas = mutableListOf<Respuesta>()
+
+    override fun analizarRespuesta(respuesta: Respuesta){
+        respuestas.add(respuesta)
+    }
+
+
+    /* Se devuelve el tiempo promedio entre las respuestas guardadas */
+    fun tiempoPromedio() = respuestas.map{it.tiempo}.average()
+
+    /* Dadas dos fechas (una de comienzo y otra de fin) se determina cuantos pedidos hubo en el medio */
+    fun cantidadDePedidosEntre(primerMomento: LocalDateTime, segundoMomento: LocalDateTime) =
+            respuestas.filter{
+                r -> r.pedido.fechaHora.isAfter(primerMomento) && r.pedido.fechaHora.isBefore(segundoMomento)
+            }.count()
```

Comment:
Podés hacer directamente el `count`:
```suggestion
            respuestas.count{
                r -> r.pedido.fechaHora.isAfter(primerMomento) && r.pedido.fechaHora.isBefore(segundoMomento)
            }
```